test/jdk/java/util/zip/ZipFile/CenSizeTooLarge.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2022, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 8272746
+ * @summary Verify that ZipFile rejects a ZIP with a CEN size which does not fit in a Java byte array
+ * @requires (sun.arch.data.model == "64")
+ * @run testng CenSizeTooLarge
+ */
+
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
```

Comment:
As this will be a new test, could you please consider converting to junit.

Reply:
- Added a comment explaining the chosen value for the constant, referencing APPNOTE.TXT 
- Converted the test to JUnit (even remembered to swap argument order of assertEquals!)